flake.nix

```diff
@@ -0,0 +1,50 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
+    purescript-overlay = {
+      url = "github:thomashoneyman/purescript-overlay";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { self, nixpkgs, ... }@inputs:
+    let
+      supportedSystems = ["x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin"];
+
+      forAllSystems = nixpkgs.lib.genAttrs supportedSystems;
+
+      nixpkgsFor = forAllSystems (system: import nixpkgs {
+        inherit system;
+        config = { };
+        overlays = builtins.attrValues self.overlays;
+      });
+    in {
+      overlays = {
+        purescript = inputs.purescript-overlay.overlays.default;
+      };
+
+      packages = forAllSystems (system:
+        let pkgs = nixpkgsFor.${system}; in {
+          default = pkgs.hello; # your package here
+        });
```

Comment:
We're only using a devshell so we can remove this.

Reply:
noticed too

commented out and 

```
 ~/projects/setup-purescript   main  nix shell
warning: Git tree '/home/srghma/projects/setup-purescript' is dirty
error: flake 'git+file:///home/srghma/projects/setup-purescript' does not provide attribute 'packages.x86_64-linux.default' or 'defaultPackage.x86_64-linux'
 ✘  ~/projects/setup-purescript   main ±  nix shell #.devShells
warning: Git tree '/home/srghma/projects/setup-purescript' is dirty
error: flake 'git+file:///home/srghma/projects/setup-purescript' does not provide attribute 'packages.x86_64-linux.default' or 'defaultPackage.x86_64-linux'
```